include/drivers/clock_control.h

```diff
@@ -270,6 +275,42 @@ static inline int clock_control_set_rate(const struct device *dev,
 	return api->set_rate(dev, sys, rate);
 }
 
+/**
+ * @brief Configure a source clock
+ *
+ * This function is non-blocking and can be called from any context.
+ * On success, the selected clock is configured as per caller's request.
+ *
+ * It is caller's responsibility to ensure that subsequent calls to the API
+ * provide the right information to enable clock_control driver to perform
+ * the right action (such as using the right clock source on _get_rate call).
+ *
+ * Argument 'data' is implementation specific and could be used to convey
+ * supplementary information required for expected clock configuration.
+ *
+ * @param dev Device structure whose driver controls the clock
+ * @param sys Opaque data representing the clock
+ * @param data Opaque data providing additional input for clock configuration
+ * @return 0 on success, negative errno on failure.
+ */
+static inline int clock_control_configure(const struct device *dev,
```

Comment:
I think we should rename this slightly to be specific to what is getting configure (ie is this just for clock source selection)

Reply:
This was initially _select() and was then updated to be more generic on this initial comment: https://github.com/zephyrproject-rtos/zephyr/pull/43790#discussion_r826718791

Reply:
@galak Also, given the discussion we had yesterday, maybe it's not bad to 'open' the API given:
- we'll likely have a clock_control overhaul in medium term. Letting experiments happen might be a good thing before and then build on these experiments
- since this api implementation is vendor specific anyway, these isn't much risk anyway

Reply:
My concern with the name is its not clear at all what is being configured an adding an API that vague doesn't make sense to me.